stdlib/src/bit/bit.mojo

```diff
@@ -289,6 +304,116 @@ fn bit_width[
         return bitwidth - leading_zero
 
 
+# ===----------------------------------------------------------------------===#
+# bit_ceil
+# ===----------------------------------------------------------------------===#
+
+
+@always_inline("nodebug")
+fn bit_ceil(val: Int) -> Int:
+    """Computes the smallest power of 2 that is greater than or equal to the
+    input value. Any integral value less than or equal to 1 be ceiled to 1.
+
+    Args:
+        val: The input value.
+
+    Returns:
+        The smallest power of 2 that is greater than or equal to the input value.
+    """
+    if val <= 1:
+        return 1
+
+    if val & (val - 1) == 0:
+        return val
+
+    return 1 << bit_width(val - 1)
+
+
+@always_inline("nodebug")
+fn bit_ceil[
+    type: DType, simd_width: Int
+](val: SIMD[type, simd_width]) -> SIMD[type, simd_width]:
+    """Computes the smallest power of 2 that is greater than or equal to the
+    input value for each element of a SIMD vector. Any integral value less than
+    or equal to 1 will be ceiled to 1.
+
+    Parameters:
+        type: `dtype` used for the computation.
+        simd_width: SIMD width used for the computation.
+
+    Constraints:
+        The element type of the input vector must be integral.
+
+    Args:
+        val: The input value.
+
+    Returns:
+        A SIMD value where the element at position `i` is the smallest power of 2
+        that is greater than or equal to the integer at position `i` of the input
+        value.
+    """
+    constrained[type.is_integral(), "must be integral"]()
+
+    alias ones = SIMD[type, simd_width].splat(1)
+
+    var less_then_one = (val <= ones).select(ones, val)
+
+    return (val > ones).select(1 << bit_width(val - ones), less_then_one)
```

Comment:
Could we simply write this?
```suggestion
	return (val <= 1).select(1, 1 << bit_width(val - 1))
```

Reply:
just fixed it, thank!

Reply:
Why do you need two `select` at all?

Reply:
ohhh my bad i misread your suggestions.

Reply:
the added line seems to cause a crash in the reverse test sometime, I still kept the `ones` and did 
```mojo
(val > 1).select(1 << bit_width(val - ones), ones)
```
to avoid the crashing.